statsd/statsd.go

```diff
@@ -552,6 +563,17 @@ func (c *Client) Gauge(name string, value float64, tags []string, rate float64)
 	return c.send(metric{metricType: gauge, name: name, fvalue: value, tags: tags, rate: rate, globalTags: c.tags, namespace: c.namespace})
 }
 
+// Gauge measures the value of a metric at a given time.
+// Even with client side aggregation enabled, there is no aggregation done on a metric
+// sent using GaugeWithTimestamp: it is written as is in the serialization buffer.
+func (c *Client) GaugeWithTimestamp(name string, value float64, tags []string, rate float64, timestamp time.Time) error {
```

Comment:
We should return an error if timestamp is 0 (same for count).

Reply:
I think it's an error since go has default value. This means an non-initialized timestamp. 
Sending a `time.Time` at 0 will produce a `Unix` timestamp of `-62135596800`.

Reply:
A gotcha... Also linked to your other question about `time.Time` vs `uint64`...
You are right I'll add a test `if timestamp.Unix() < 0`, WDYT?

Reply:
Yeah.. We have to draw a limit somewhere in the clients and since the intake has a "minimum timestamp" limit and we already have other places limiting to the unix epoch in the Agent, I think having `int(0)` as the minimal possible value is speaking to everyone in a client. Let me know if you think I'm wrong, really open to discussion with these topics.

> I assume this doesn't apply to us?

It doesn't indeed, a future PR will update the public documentation, with things more related about the "features" than the technical chunks in the client.

Reply:
> I'm trying to understand why 0 (jan 1970) is the limit here.

The idea is to provide a hard-coded limit instead of no limit, even if it has no actual feature meaning here.
I added a mention in the comment to redirect to the feature documentation.
```
Please report to the Datadog documentation for the maximum age of a metric.
```